core/sdk-core/src/main/java/software/amazon/awssdk/core/async/AsyncRequestBody.java

```diff
@@ -405,34 +406,36 @@ static AsyncRequestBody empty() {
 
     /**
      * Converts this {@link AsyncRequestBody} to a publisher of {@link AsyncRequestBody}s, each of which publishes a specific
-     * portion of the original data, based on the configured {code chunkSizeInBytes}.
+     * portion of the original data, based on the provided {@link AsyncRequestBodySplitConfiguration}. The default chunk size
+     * is 2MB and the default buffer size is 8MB.
      *
      * <p>
      * If content length of this {@link AsyncRequestBody} is present, each divided {@link AsyncRequestBody} is delivered to the
      * subscriber right after it's initialized.
      * <p>
-     * // TODO: API Surface Area review: should we make this behavior configurable?
      * If content length is null, it is sent after the entire content for that chunk is buffered.
      * In this case, the configured {@code maxMemoryUsageInBytes} must be larger than or equal to {@code chunkSizeInBytes}.
      *
-     * @param chunkSizeInBytes      the size for each divided chunk. The last chunk may be smaller than the configured size.
-     * @param maxMemoryUsageInBytes the max memory the SDK will use to buffer the content
-     * @return SplitAsyncRequestBodyResult
+     * @see AsyncRequestBodySplitConfiguration
      */
-    default SdkPublisher<AsyncRequestBody> split(long chunkSizeInBytes, long maxMemoryUsageInBytes) {
-        Validate.isPositive(chunkSizeInBytes, "chunkSizeInBytes");
```

Comment:
Where are we doing these validations now ?
Should it be done while building `AsyncRequestBodySplitConfiguration` ?

Reply:
Yeah, the validation is in `AsyncRequestBodySplitConfiguration` now. https://github.com/aws/aws-sdk-java-v2/pull/4236/files#diff-0ad88dc6bfa931f7dd9a3eeb93928f422634f321f7d63939bb6a20e5f4752c21R34